front/app/containers/Admin/projects/project/analysis/Insights/index.tsx

```diff
@@ -23,13 +24,25 @@ const Insights = () => {
   const { data: insights, isLoading } = useAnalysisInsights({
     analysisId,
   });
+
+  const largeSummariesEnabled = useFeatureFlag({
+    name: 'large_summaries',
+    onlyCheckAllowed: true,
+  });
+
   const filters = useAnalysisFilterParams();
-  const { data } = useInfiniteAnalysisInputs({
+
+  const { data: allInputs } = useInfiniteAnalysisInputs({
+    analysisId,
+  });
+  const { data: filteredInputs } = useInfiniteAnalysisInputs({
     analysisId,
     queryParams: filters,
   });
 
-  const inputsCount = data?.pages[0].meta.filtered_count;
+  const inputsCount = allInputs?.pages[0].meta.filtered_count || 0;
+  const filteredInputsCount = filteredInputs?.pages[0].meta.filtered_count || 0;
+  const applyInputsLimit = !largeSummariesEnabled && filteredInputsCount > 30;
```

Comment:
I'm not sure I'm following, but isn't the `largeSummariesEnabled` feature also checked in the case of Q&A here? Shouldn't it only be used for summaries?